File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/cart/entity/CartItems.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.cart.entity;

import com.multicampus.gamesungcoding.a11ymarketserver.common.id.UuidV7;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.entity.Product;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.util.UUID;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class CartItems {

    @Id
    @UuidV7
    @Column(length = 16, updatable = false, nullable = false)
    private UUID cartItemId;                 // 장바구니 아이템 PK

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cart_id", updatable = false, nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Cart cart;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", updatable = false, nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Product product;

    @Column(nullable = false)
    private int quantity;

    @Builder
    private CartItems(Cart cart, Product product, int quantity) {
        this.cart = cart;
        this.product = product;
        this.quantity = quantity;
    }

    public void changeQuantity(int quantity) {
        if (quantity < 1) {
            throw new IllegalArgumentException("quantity must be >= 1");
        }
        this.quantity = quantity;
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/cart/entity/CartItems.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.cart.entity

import com.multicampus.gamesungcoding.a11ymarketserver.common.id.UuidV7
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.entity.Product
import jakarta.persistence.*
import lombok.AccessLevel
import lombok.Getter
import lombok.NoArgsConstructor
import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction
import java.util.*

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
class CartItems
    (
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "cart_id", updatable = false, nullable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    var cart: Cart,

    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "product_id", updatable = false, nullable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    var product: Product?,

    @Column(nullable = false)
    var quantity: Int
) {
    @Id
    @UuidV7
    @Column(length = 16, updatable = false, nullable = false)
    private var cartItemId: UUID? = null // 장바구니 아이템 PK

    fun changeQuantity(quantity: Int) {
        require(quantity >= 1) { "quantity must be >= 1" }
        this.quantity = quantity
    }
}